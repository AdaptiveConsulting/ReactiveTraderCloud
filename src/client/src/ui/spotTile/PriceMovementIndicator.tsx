import * as React from 'react'
import * as classnames from 'classnames'
import './priceMovement.scss'

const priceMovementTypes = {
  Up: 'Up',
  Down: 'Down',
}

export interface PriceMovementIndicatorProps {
  priceMovementType: any
  spread: {
    formattedValue: string,
  }
}

const getCaretDirection = (priceMovement: string) => {

  const upDirectionClasses = {
    [priceMovementTypes.Up]: 'fa-caret-up',
    [priceMovementTypes.Down]: 'price-movement__icon--inactive',
  }

  const downDirectionClasses = {
    [priceMovementTypes.Down]: 'fa-caret-down',
    [priceMovementTypes.Up]: 'price-movement__icon--inactive',
  }

  'price-movement__icon--up fa fa-lg fa-caret-up'

  return {
    up: classnames('price-movement__icon--up fa fa-lg', upDirectionClasses[priceMovement]),
    down: classnames('price-movement__icon--down fa fa-lg', downDirectionClasses[priceMovement]),
  }
}

// tslint:disable-next-line:variable-name
const PriceMovementIndicator = (props: PriceMovementIndicatorProps) => {

  const { priceMovementType, spread } = props
  const priceMovementClassNames = getCaretDirection(priceMovementType.name)
  return (
    <div>
      <div className="price-movement">
        <i className={priceMovementClassNames.up}></i>
        <span className="price-movement__value">{spread.formattedValue}</span>
        <i className={priceMovementClassNames.down}></i>
      </div>
    </div>
  )
}

export default PriceMovementIndicator
