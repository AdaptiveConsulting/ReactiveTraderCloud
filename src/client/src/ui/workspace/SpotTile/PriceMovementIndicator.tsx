import * as React from 'react'
import * as classnames from 'classnames'
import './priceMovement.scss'

const priceMovementTypes = {
  Up: 'Up',
  Down: 'Down',
}

export interface PriceMovementIndicatorProps {
  priceMovementType: string
  spread: {
    formattedValue: string,
  }
}

const PriceMovementIndicator = (props: PriceMovementIndicatorProps) => {

  const { priceMovementType, spread } = props

  const upDirectionClassName = classnames(
    'price-movement__icon--up fa fa-lg',
    {
      'fa-caret-up': priceMovementType === priceMovementTypes.Up,
    },
    {
      'price-movement__icon--inactive': priceMovementType === priceMovementTypes.Down,
    },
  )

  const downDirectionClassName = classnames(
    'price-movement__icon--down fa fa-lg',
    {
      'fa-caret-down': priceMovementType === priceMovementTypes.Down,
    },
    {
      'price-movement__icon--inactive': priceMovementType === priceMovementTypes.Up,
    },
  )

  return (
    <div>
      <div className="price-movement">
        <i className={upDirectionClassName}></i>
        <span className="price-movement__value">{spread.formattedValue}</span>
        <i className={downDirectionClassName}></i>
      </div>
    </div>
  )
}

export default PriceMovementIndicator
