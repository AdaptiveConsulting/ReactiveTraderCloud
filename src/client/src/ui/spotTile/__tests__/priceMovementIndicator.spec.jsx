import * as React from 'react'
import PriceMovementIndicator from '../PriceMovementIndicator'
import ShallowRenderer from 'react-test-renderer/shallow'
import { mount } from 'enzyme'

describe('PriceMovementIndicator', () => {
  test('renders correct markup Down', () => {
    const component = <PriceMovementIndicator
                    priceMovementType={'Down'}
                    spread={{
                      formattedValue: '-1.23',
                    }}/>
    const shallowRenderer = new ShallowRenderer()
    shallowRenderer.render(component)
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
  })

  test('renders correct markup Up', () => {
    const component = <PriceMovementIndicator
                    priceMovementType={'Up'}
                    spread={{
                      formattedValue: '-1.23',
                    }}/>
    const shallowRenderer = new ShallowRenderer()
    shallowRenderer.render(component)
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
  })
})
