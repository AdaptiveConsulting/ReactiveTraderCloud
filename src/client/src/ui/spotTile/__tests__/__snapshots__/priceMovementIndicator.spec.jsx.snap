// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceMovementIndicator renders correct markup Down 1`] = `
<div>
  <div
    className="price-movement"
  >
    <i
      className="price-movement__icon--up fa fa-lg"
    />
    <span
      className="price-movement__value"
    >
      -1.23
    </span>
    <i
      className="price-movement__icon--down fa fa-lg"
    />
  </div>
</div>
`;

exports[`PriceMovementIndicator renders correct markup Up 1`] = `
<div>
  <div
    className="price-movement"
  >
    <i
      className="price-movement__icon--up fa fa-lg"
    />
    <span
      className="price-movement__value"
    >
      -1.23
    </span>
    <i
      className="price-movement__icon--down fa fa-lg"
    />
  </div>
</div>
`;
